fragmenter/fragment.py

```diff
@@ -824,1078 +658,459 @@ class WBOFragmenter(Fragmenter):
+
+        atom_indices = {get_atom_index(molecule, atom) for atom in atoms}
+
+        atoms_to_add = [
+            (atom_index, neighbour.molecule_atom_index)
+            for atom_index in atom_indices
+            for neighbour in molecule.atoms[atom_index].bonded_atoms
+            if neighbour.atomic_number != 1
+            and neighbour.molecule_atom_index not in atom_indices
+        ]
+        map_atoms_to_add = [
+            (
+                get_map_index(molecule, j),
+                (get_map_index(molecule, i), get_map_index(molecule, j)),
+            )
+            for i, j in atoms_to_add
+        ]
+
+        # Compute the distance from each neighbouring atom to each of the atoms in the
+        # target bond.
+        nx_molecule = molecule.to_networkx()
+
+        target_indices = [get_atom_index(molecule, atom) for atom in target_bond]
+
+        path_lengths_1, path_lengths_2 = zip(
+            *(
+                (
+                    networkx.shortest_path_length(
+                        nx_molecule, target_index, neighbour_index
+                    )
+                    for target_index in target_indices
+                )
+                for atom_index, neighbour_index in atoms_to_add
+            )
+        )
+
+        if len(path_lengths_1) == 0 and len(path_lengths_2) == 0:
+            return None
+
+        reverse = False
+
+        min_path_length_1 = min(path_lengths_1)
+        min_path_length_2 = min(path_lengths_2)
+
+        if min_path_length_1 < min_path_length_2:
+            sort_by = path_lengths_1
+        elif min_path_length_2 < min_path_length_1:
+            sort_by = path_lengths_2
 
-        Parameters
-        ----------
-        bond: tuple of ints
-            map indices of atoms in bond
+        else:
 
-        Returns
-        -------
-        atoms, bonds: set of ints (map indices of atoms in quartet), set of tuples of ints (bonds in quartet)
+            # If there are multiple neighbouring atoms the same path length away
+            # from the target bond fall back to sorting by the WBO.
+            map_atoms_to_add = [
+                map_tuple
+                for map_tuple, *path_length_tuple in zip(
+                    map_atoms_to_add, path_lengths_1, path_lengths_2
+                )
+                if min_path_length_1 in path_length_tuple
+            ]
+
+            sort_by = [
+                molecule.get_bond_between(
+                    get_atom_index(molecule, neighbour_bond[0]),
+                    get_atom_index(molecule, neighbour_bond[1]),
+                ).fractional_bond_order
+                for _, neighbour_bond in map_atoms_to_add
+            ]
+
+            reverse = True
+
+        sorted_atoms = [
+            a for _, a in sorted(zip(sort_by, map_atoms_to_add), reverse=reverse)
+        ]
+
+        return None if len(sorted_atoms) == 0 else sorted_atoms[0]
+
+    @classmethod
+    def _select_neighbour_by_wbo(
```

Comment:
You don't actually use `cls` here or in the other classmethod, could it be a separate utility or a staticmethod?

Reply:
I typically mark methods which users may override in a sub-class as a `classmethod` so that the can easily call `super` if needed. I think whether inheritance or composition is the best way to customise this behaviour is beyond the scope of this PR however, but please feel free to open an issue if you'd like to discuss this further!